ngraph/core/reference/src/runtime/reference/pad.cpp

```diff
@@ -191,8 +191,8 @@ namespace ngraph
                                 }
                                 else
                                 {
-                                    c[i] = static_cast<size_t>(padding_below[i] + src_dim +
-                                                               padding_above[i] - pos);
+                                    c[i] = static_cast<size_t>(2 * (padding_below[i] + src_dim) -
```

Comment:
I am not sure that I get this code right. Does this change contradicts with the specification where we have an example with `pads_begin[i] != pads_end[i]`. Could you elaborate more on why you changed the reference implementation?
The example is the following

```
pads_begin = [0, 1]
pads_end = [2, 3]
```

```
INPUT =
[[ 1  2  3  4 ]
[  5  6  7  8 ]
[  9 10 11 12 ]]
```

```
OUTPUT =
[[ 1  1  2  3  4  4  3  2 ]
[  5  5  6  7  8  8  7  6 ]
[  9  9 10 11 12 12 11 10 ]
[  9  9 10 11 12 12 11 10 ]
[  5  5  6  7  8  8  7  6 ]]
```

Reply:
Consider new example:
```
pads_begin = [0, 0]
pads_end = [1, 1]
```

```
INPUT =
[[ 1  2  3  4 ]
[  5  6  7  8 ]
[  9 10 11 12 ]]
```

```
OUTPUT =
[[ 1  2  3  4  4 ]
[  5  6  7  8  8 ]
[  9 10 11 12 12 ]
[  9 10 11 12 12 ]]
```
Find new indexes for value with coords [0, 4] (in example its must be value 4 with src indexes [0, 3]):
`i = 1  // (second dim)`
`179: pos = padding_below[i] - (c[i] + 1);  // pos = 0 - (4 + 1) = -5`
`186: pos = -(pos + 1);  // pos = 4`
`194: c[i] = static_cast<size_t>(padding_below[i] + src_dim + padding_above[i] - pos);  // c[i] = (0 + 4 + 1 - 4) = 1`

We get 1 but must get 3.

New formula:
`2 * (padding_below[i] + src_dim) - c[i] - 1  // c[1] = (2 * (0 + 4)  - 4 - 1 = 3`
We get 3 and its right.

Reply:
I don't get an answer for my question :)
Does this change contradict with the specification?

Reply:
Have a chat and got an explanation.